### PR TITLE
UP-4964:  Fix configuration of BranchingRenderingPipeline on uP5

### PR DESCRIPTION
**Pull request by drewwills**
https://issues.jasig.org/browse/UP-4964

##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] tests are included
-   [x] documentation is changed or added

##### Description of change

The configuration patterns we adopted for uP5 are a big improvement.  Nevertheless, some uPortal capabilities are not yet manageable in the uP5 way.  One such capability is uPortal Rendering Pipeline customization based on branching.  This PR aims to address this bug -- to make it possible to configure branching pipeline settings properly in uPortal-start.

I intend to add documentation for these features before removing the `[WIP]` tag.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
AppVeyor is failing due to unrelated issue from https://issues.jasig.org/browse/UP-4965

File: uPortal-rendering/src/main/java/org/apereo/portal/rendering/RenderingPipelineBranchPoint.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    private int order = Integer.MAX_VALUE; // default -- last position
+
+    // dependency-injected
+    private Predicate<HttpServletRequest> predicate;
+
+    // dependency-injected
+    private IPortalRenderingPipeline alternatePipe;
+
+    public void setOrder(int order) {
+        this.order = order;
```
Comment (ChristianMurphy) [inline uPortal-rendering/src/main/java/org/apereo/portal/rendering/RenderingPipelineBranchPoint.java]: :question: what happens if two branch points have the same `order`?

Comment (drewwills) [reply uPortal-rendering/src/main/java/org/apereo/portal/rendering/RenderingPipelineBranchPoint.java]: One of them (randomly selected) will be processed first.  IMHO that's fine.  The alternative would be detecting the circumstance and throwing an exception.

Comment (drewwills) [reply uPortal-rendering/src/main/java/org/apereo/portal/rendering/RenderingPipelineBranchPoint.java]: Perhaps logging a warning is a good middle ground.

**Pull request merged**